### PR TITLE
[FIX] base_automation: warn email are not compatible with unlink

### PR DESCRIPTION
**Pull request by Julien00859**
Create a automated action on deletion that send an email, delete a
related record. The email is not sent.

Email are linked to their chatter message, when the later is deleted the
former is deleted in cascade too. This is a known limitation of the mail
model.

Task: 2151519

### CODE REVIEW

**Comment by Julien00859** [comment]
@fw-bot up to saas-13.2

**Comment by Julien00859** [comment]
We should find a solution on master

**Comment by Julien00859** [comment]
@fx-bot up to master
@Xavier-Do 
Workaround is to change the action to a "execute python code" that creates an mail.mail. Changing the default behavior of mail.message so it is not attached to the to-be-unlinked record is too much sensitive even for master.

**Comment by Julien00859** [comment]
@fw-bot up to master

File: addons/base_automation/models/base_automation.py [hunk 1]
```diff
@@ -91,6 +91,18 @@ def _onchange_state(self):
...
 
+        mail_states = {'email', 'followers', 'next_activity'}
+        if self.trigger == 'on_unlink' and self.state in mail_states:
+            ff = self.fields_get(['trigger', 'state'])
+            return {'warning': {
+                'title': _("Warning"),
+                'message': _("The \"%(trigger_value)s\" %(trigger_label)s can not be used with the \"%(state_value)s\" action type") % {
+                    'trigger_value': dict(ff['trigger']['selection'])['on_unlink'],
+                    'trigger_label': ff['trigger']['string'],
+                    'state_value': dict(ff['state']['selection'])[self.state],
```
Comment (rco-odoo) [inline addons/base_automation/models/base_automation.py:102]: That's much work for
```suggestion
                'message': _(
                    "You cannot send an email, add followers or create an activity "
                    "for a deleted record.  It simply does not work."
                ),
```

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**